### PR TITLE
[OTJ] Implementing "saddle" mechanic

### PR DESCRIPTION
**Pull request by theelk801**
I think this is ready, there's more functionality to add for other cards but that doesn't need to be done in this PR

### CODE REVIEW

File: Mage/src/main/java/mage/game/events/GameEvent.java [hunk 1]
```diff
@@ -176,6 +176,24 @@ combines all player life lost events to a single batch (event)
          sourceId    sourceId of the vehicle
          playerId    the id of the controlling player
          */
+        SADDLE_MOUNT,
+        /* SADDLE_MOUNT
+         targetId    the id of the creature that saddled a mount
```
Comment (xenohedron) [inline Mage/src/main/java/mage/game/events/GameEvent.java:181]: past tense description but this is a present tense event

File: Mage/src/main/java/mage/abilities/keyword/SaddleAbility.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        game.fireEvent(GameEvent.getEvent(
+                GameEvent.EventType.MOUNT_SADDLED,
+                source.getSourceId(),
+                source, source.getControllerId()
+        ));
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Optional.ofNullable(source.getSourcePermanentOrLKI(game))
```
Comment (xenohedron) [inline Mage/src/main/java/mage/abilities/keyword/SaddleAbility.java:88]: why is this LKI ?

File: Mage/src/main/java/mage/abilities/condition/common/SaddledCondition.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * @author TheElk801
+ */
+public enum SaddledCondition implements Condition {
+    instance;
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return Optional
+                .ofNullable(source.getSourcePermanentIfItStillExists(game))
+                .filter(Objects::nonNull)
```
Comment (xenohedron) [inline Mage/src/main/java/mage/abilities/condition/common/SaddledCondition.java:21]: as previously discussed, pointless to filter null on optionals

File: Mage.Tests/src/test/java/org/mage/test/cards/abilities/keywords/SaddleTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        );
+    }
+
+    @Test
+    public void testNoSaddle() {
+        addCard(Zone.BATTLEFIELD, playerA, arynx);
+
+        attack(1, playerA, arynx, playerB);
+
+        setStopAt(1, PhaseStep.END_TURN);
```
Comment (xenohedron) [inline Mage.Tests/src/test/java/org/mage/test/cards/abilities/keywords/SaddleTest.java:33]: miss strict choose mode

**Review by xenohedron — APPROVED** [review]

File: Mage/src/main/java/mage/abilities/keyword/SaddleAbility.java (hunk 2 above)
Comment (theelk801) [reply Mage/src/main/java/mage/abilities/keyword/SaddleAbility.java:88]: just a mistake, thanks

**Pull request merged**